docker/test/container_release_image_test.go

```diff
@@ -60,53 +91,122 @@ func TestCommitterReleaseImagesWithTLS(t *testing.T) {
+						)
+					}
+
+					// start a secured database node and return the db password.
+					params.dbPassword = startSecuredDatabaseNode(ctx, t, params.asNode(dbNode))
+					// start the orderer node.
+					startCommitterNodeWithTestImage(ctx, t, params.asNode(ordererNode))
+					// start the committer nodes.
+					for _, node := range committerNodes {
+						startCommitterNodeWithReleaseImage(ctx, t, params.asNode(node))
+					}
+					// start the load generator node.
+					startLoadgenNodeWithReleaseImage(ctx, t, params.asNode(loadgenNode))
+
+					monitorMetric(t,
+						getContainerMappedHostPort(
+							ctx, t, assembleContainerName("loadgen", mode, dbType), loadGenMetricsPort,
+						),
+					)
+				})
 			}
-			monitorMetric(t,
-				getContainerMappedHostPort(ctx, t, assembleContainerName("loadgen", mode), loadGenMetricsPort),
-			)
 		})
 	}
 }
 
+// CreateAndStartSecuredDatabaseNode creates a containerized YugabyteDB or PostgreSQL
+// database instance in a secure mode.
+func startSecuredDatabaseNode(ctx context.Context, t *testing.T, params startNodeParameters) string {
+	t.Helper()
+
+	tlsConfig, _ := params.credsFactory.CreateServerCredentials(t, params.tlsMode, params.node)
+
+	node := &dbtest.DatabaseContainer{
+		DatabaseType: params.dbType,
+		Network:      params.networkName,
+		Hostname:     params.node,
+		DbPort:       defaultDBPort,
+		TLSConfig:    &tlsConfig,
+	}
+
+	node.StartContainer(ctx, t)
+	t.Cleanup(func() {
+		node.StopAndRemoveContainer(t)
+	})
+	conn := node.GetConnectionOptions(ctx, t)
+
+	// This is relevant if a different CA was used to issue the DB's TLS certificates.
+	require.NotEmpty(t, tlsConfig.CACertPaths)
+	conn.TLS = connection.DatabaseTLSConfig{
+		Mode:       connection.OneSideTLSMode,
+		CACertPath: tlsConfig.CACertPaths[0],
+	}
+
+	// post start container tweaking
+	switch node.DatabaseType {
+	case testutils.YugaDBType:
+		// Must run after node startup to ensure proper root ownership and permissions for the TLS certificate files.
+		_, exitCode := node.ExecuteCommand(t, []string{
+			"chown", "root:root",
+			filepath.Join("/creds", dbtest.YugabytePublicKeyFileName),
+			filepath.Join("/creds", dbtest.YugabytePrivateKeyFileName),
+		})
+		require.Zero(t, exitCode)
+		node.EnsureNodeReadinessByLogs(t, dbtest.YugabytedReadinessOutput)
+		conn.Password = node.ReadPasswordFromContainer(t, containerPathForYugabytePassword)
+	case testutils.PostgresDBType:
+		// Must run after node startup to ensure proper root ownership and permissions for the TLS certificate files.
+		_, exitCode := node.ExecuteCommand(t, []string{
+			"chown", "postgres:postgres",
+			filepath.Join("/creds", dbtest.PostgresPublicKeyFileName),
+			filepath.Join("/creds", dbtest.PostgresPrivateKeyFileName),
+		})
+		require.Zero(t, exitCode)
+		node.EnsurePostgresNodeReadiness(t, "5433")
+		_, exitCode = node.ExecuteCommand(t, append(enforcePostgresSSLScript, reloadPostgresConfigScript...))
```

Comment:
minor: These are always used as a single command. I suggest merging them to a single variable.